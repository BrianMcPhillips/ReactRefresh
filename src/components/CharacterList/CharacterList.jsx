import React from 'react';
import PropTypes from 'prop-types';
import Character from '../Character/Character';

const CharacterList = ({ data }) => {
  const characterStuff = data.map(character => (
    <li key={character._id}>
      <Character {...character}/>
    </li>
  ));

  return (
    <ul>
      {characterStuff}
    </ul>
  );
};

CharacterList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    _id: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
