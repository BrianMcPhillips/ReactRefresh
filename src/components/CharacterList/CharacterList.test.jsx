import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from "react-router-dom";

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders CharacterList', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList 
          data={[
            {
              name: 'Test',
              image: 'http://placecage.com/200/200',
              _id: '12345'
            }
          ]}
        />
      </MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
