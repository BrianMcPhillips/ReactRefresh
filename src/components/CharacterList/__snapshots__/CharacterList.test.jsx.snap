// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders CharacterList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <div>
        <a
          href="/detail/12345"
        >
          <img
            alt="Test"
            src="http://placecage.com/200/200"
          />
        </a>
        <h2>
          Test
        </h2>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
